File: Pilha/PilhaJava/PilhaEstatica/PilhaDupla/DuplamenteEmpilhavel.java
package Pilha.PilhaJava.PilhaEstatica.PilhaDupla;

public interface DuplamenteEmpilhavel {

    //Metodos auxiliares
    public boolean estaVazia1();
    public boolean estaVazia2();
    public boolean estaCheia1();
    public boolean estaCheia2();
    public String imprimir1();
    public String imprimir2();

    //metodos principais
    public void empilhar1(Object dado1);
    public void empilhar2(Object dado2);
    public Object desempilhar1();
    public Object desempilhar2();
    public Object topo1();
    public Object topo2();
}

File: Pilha/PilhaKotlin/PilhaEstatica/PilhaDupla/DuplamenteEmpilhavel.kt
interface DuplamenteEmpilhavel {
    // Metodos principais
    fun empilhar1(dado1: Any?)
    fun empilhar2(dado2: Any?)
    fun desempilhar1(): Any?
    fun desempilhar2(): Any?
    fun topo1(): Any?
    fun topo2(): Any?
    // Metodos auxiliares
    fun estaCheia1(): Boolean
    fun estaCheia2(): Boolean
    fun estaVazia1(): Boolean
    fun estaVazia2(): Boolean
    fun imprimir1(): String
    fun imprimir2(): String
}
